core/fedavgoptimizer.py

```diff
@@ -0,0 +1,144 @@
+from core.utils.event_types import ActionableEventTypes, RawEventTypes, callback_handler
+from core.utils.dmljob import deserialize_job
+from core.utils.dmlresult import DMLResult
+
+class FederatedAveragingOptimizer(object):
+	"""
+	Federated Averaging Optimizer
+
+	Implements the Federated Averaging algorithm (with some tweaks)
+	and acts as a "brain" for the Unix Service
+
+	"""
+	def __init__(self, content):
```

Comment:
Can you rename "content" to something more descriptive? "content" is too general.

Reply:
In fact, it is the result of doing a .get("content") on what is passed in.

Reply:
CommMgr does .get("content") and passes that into the optimizer. So this is, literally, content.

Reply:
Please don't resolve the conversation if you didn't initialize it, as they got lost without actually being solved.

I'm not convinced with your answer. How about renaming it to something like "initialization_payload"?

Reply:
No, it's named this way so that you can traceback where the argument is coming from.
The 'init' method doesn't need to have an argument with the name 'initialization' used to make it more descriptive